sourcerer-dbstore/src/main/java/org/elder/sourcerer2/dbstore/DbstoreEventStore.kt

```diff
@@ -0,0 +1,50 @@
+package org.elder.sourcerer2.dbstore
+
+import org.elder.sourcerer2.StreamId
+
+
+interface DbstoreEventStore {
+    /**
+     * The number of independent shards that this repository supports. Each individual event stream will be
+     * automatically assigned to a shard when the stream is first created in the range of [0,shards). The number of
+     * shards can be configured to go up over time, but not down (or some events would be excluded). Note that as
+     * shards are assigned when the stream is first created, increasing the number of shards at a later stage will not
+     * re-balance old streams, it will only ensure that newly created streams are spread across all then available
+     * shards.
+     */
+    val shards: Int
+
+    /**
+     * Reads events from a given particular point in time (version) across all of the events in this repository.
+     * @param shard the shard to read from, if null, events will be read across all of the shards (in order).
+     */
+    fun readRepositoryEvents(
+            category: String,
+            shard: Int?,
+            fromVersion: DbstoreRepositoryVersion? = null,
+            maxEvents: Int = 4096
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.elder.org/coding_rules#rule_key=detekt-kotlin%3AMagicNumber)